test/jdk/java/awt/MouseInfo/ContainerMousePositionTest.java

```diff
@@ -0,0 +1,141 @@
+            Point pMousePosition = frame.getMousePosition(false);
+            if (pMousePosition != null) {
+                throw new RuntimeException("Test failed: Container is overlapped by " + " child and it should be taken into account");
+            }
+            System.out.println("Test stage completed: Container is overlapped by " + " child and it was taken into account");
+
+            pMousePosition = frame.getMousePosition(true);
+            if (pMousePosition == null) {
+                throw new RuntimeException("Test failed: Container is overlapped by " + " child and it should not be taken into account");
+            }
+            System.out.println("Test stage completed: Container is overlapped by " + " child and it should not be taken into account");
+
+            robot.mouseMove(panel.getLocationOnScreen().x + POINT_WITHOUT_COMPONENTS.x, panel.getLocationOnScreen().y + POINT_WITHOUT_COMPONENTS.y);
+
+            robot.delay(DELAY);
+
+            pMousePosition = panel.getMousePosition(true);
+            if (pMousePosition == null) {
+                throw new RuntimeException("Test failed: Pointer was outside of " + "the component so getMousePosition() should not return null");
+            }
+            System.out.println("Test stage completed: Pointer was outside of " + "the component and getMousePosition() has not returned null");
+
+            pMousePosition = panel.getMousePosition(false);
+            if (pMousePosition == null) {
+                throw new RuntimeException("Test failed: Pointer was outside of " + "the component so getMousePosition() should not return null");
+            }
+            System.out.println("Test stage completed: Pointer was outside of " + "the component and getMousePosition() has not returned null");
+
+            robot.mouseMove(frame.getLocationOnScreen().x + frame.getWidth() + POINT_WITHOUT_COMPONENTS.x, frame.getLocationOnScreen().y + frame.getHeight() + POINT_WITHOUT_COMPONENTS.y);
+
+            robot.delay(DELAY);
+
+            pMousePosition = frame.getMousePosition(true);
+            if (pMousePosition != null) {
+                throw new RuntimeException("Test failed: Pointer was outside of " + "the Frame widow and getMousePosition() should return null");
+            }
+            System.out.println("Test stage completed: Pointer was outside of " + "the Frame widow and getMousePosition() returned null");
+
+            pMousePosition = frame.getMousePosition(false);
+            if (pMousePosition != null) {
+                throw new RuntimeException("Test failed: Pointer was outside of " + "the Frame widow and getMousePosition() should return null");
+            }
+            System.out.println("Test stage completed: Pointer was outside of " + "the Frame widow and getMousePosition() returned null");
+
+            robot.delay(DELAY);
+
+            System.out.println("ComponentMousePositionTest PASSED.");
+        } finally {
+            if (frame != null) {
```

Comment:
frame var is assigned on EDT and then used on non-EDT thread.

Reply:
Updated.